CMakeLists.txt

```diff
@@ -184,6 +196,9 @@ option(BUILD_INSTALLER "Build installer" OFF)
 option(BUILD_TESTS "Build tests" ON)
 option(USE_GIT_COMMIT "Show the git commit in version information" ON)
 option(REQUIRE_PTHREADS "For non-Windows systems, set whether Csound will use threads or not" ON)
+option(DEBUG_ERROR_ON_WARNING "Error on compiler warning" ON)
+option(APPLE_FRAMEWORK "Build framework on Apple" ON)
+option(USE_VCPKG_DEPENDENCIES "Use dependencies from vcpkg" OFF)
```

Comment:
There is already a USE_VCPKG no? What is this flag supposed to do

Reply:
This is for building for the vcpkg build bots, not for vcpkg users. I guess I could change the name? We might be able to merge the two flags at some point?

Reply:
I don't quite get the comment about vcpkg build bots and vcpkg users. What do you mean? They are the same flag but I can see now that the `USE_VCPKG` flag is not referenced much in this branch.

I think trying to add this work to the older Csound6 branch is going to be a bit of a mess. Why not try this with develop? That has a working build for VCPKG for Linux and Windows at least, I think Mac is also working.

Reply:
Maybe that would make sense. I'm just a little tentative about using an unreleased version?

Reply:
The difference is the difference between `./vcpkg install csound` (where vcpkg does the building for you) and `cmake ... -DCMAKE_TOOLCHAIN_FILE=[vcpkg root]/scripts/buildsystems/vcpkg.cmake` (where users manually call cmake). But there should be a significant amount of overlap in the changes needed for both for sure. But yeah, I'll try working with the develop branch next

Reply:
Right I see what you're referring to. I think that the vcpkg build for Csound will just be a certain build configuration for the CMake build i.e. probably without any unsupported flags that require 3rd party deps not in vcpkg rather than a separate build flow in CMake. As for the unreleased version, I'm sure we can work around that for now. It is possible to use branch/unreleased versions in VCPKG afaik but they just don't encourage it.